src/pyupgrade/plugins/format_specififiers.rs

```diff
@@ -146,16 +147,16 @@ pub fn format_specifiers(checker: &mut Checker, expr: &Expr, func: &Expr) {
                         Some(item) => item,
                     };
                     println!("Checkpoint 3");
-                    let mut check =
-                        Check::new(CheckKind::FormatSpecifiers, Range::from_located(expr));
-                    if checker.patch(check.kind.code()) {
-                        check.amend(Fix::replacement(
+                    let mut diagnostic =
+                        Diagnostic::new(violations::FormatSpecifiers, Range::from_located(expr));
+                    if checker.patch(diagnostic.kind.code()) {
+                        diagnostic.amend(Fix::replacement(
                             new_call,
                             expr.location,
                             expr.end_location.unwrap(),
                         ));
                     }
-                    checker.add_check(check);
+                    checker.diagnostics.push(diagnostic);
```

Comment:
Thank you and sorry that this all churned mid-PR for you.

Reply:
Im never going to complain about you making this project easier to contribute to.